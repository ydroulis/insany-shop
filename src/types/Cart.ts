
export type CartProduct = {
    id: number
    name: string
    description: string
    price: number
    image: string
    stock: number
    items?: number
}

export type Cart = {
    content: CartProduct[]
    total: number
}