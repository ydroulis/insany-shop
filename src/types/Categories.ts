export type Categories = Category[];

export type Category = {
    id: string,
    name: string,
    description: string,
    icon: string,
    productCount: number
};