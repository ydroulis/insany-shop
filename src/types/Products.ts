export type Products = Product[]

export type Product = {
    id: number,
    name: string,
    description: string,
    price: number,
    image: string,
    category: string,
    stock: number,
    rating: number,
    brand?: string

}