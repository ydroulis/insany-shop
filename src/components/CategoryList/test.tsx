import React from "react";
import { render, screen, within } from "@testing-library/react";
import CategoriesList from "./";
import { Categories } from "@/types/Categories";


describe("<CategoryList />", () => {
    const categories: Categories = [
        {
            id: "1", name: "Eletrônicos", productCount: 5,
            description: "",
            icon: ""
        },
        {
            id: "2", name: "Livros", productCount: 1,
            description: "",
            icon: ""
        },
    ];

    it("should render section title with proper id", () => {
        render(<CategoriesList categories={categories} />);
        const title = screen.getByRole("heading", { name: /principais categorias/i });
        expect(title).toBeInTheDocument();
        expect(title).toHaveAttribute("id", "categories-title");
    });

    it("should associate aria-labelledby to the container of the section", () => {
        render(<CategoriesList categories={categories} />);
        const region = screen.getByRole("region");
        expect(region).toHaveAttribute("aria-labelledby", "categories-title");
    });

    it("should render a list with category list", () => {
        render(<CategoriesList categories={categories} />);
        const list = screen.getByRole("list");
        expect(list).toBeInTheDocument();

        const items = within(list).getAllByRole("listitem");
        expect(items).toHaveLength(categories.length);
    });

    it("should render CategoryCards properly", () => {
        render(<CategoriesList categories={categories} />);
        const cards = screen.getAllByRole("link");
        expect(cards).toHaveLength(categories.length);

        expect(screen.getByText("Eletrônicos")).toBeInTheDocument();
        expect(screen.getByText("5 produtos")).toBeInTheDocument();
        expect(screen.getByText("Livros")).toBeInTheDocument();
        expect(screen.getByText("1 produto")).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        const { container } = render(<CategoriesList categories={categories} />);
        expect(container).toMatchSnapshot();
    });
});
