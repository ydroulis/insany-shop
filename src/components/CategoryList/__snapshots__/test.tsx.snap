// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<CategoriesList /> should match snapshot with categories 1`] = `
<div>
  <section
    aria-labelledby="categories-title"
    class="sc-bRKDuR GtvKZ"
    role="region"
  >
    <h2
      class="sc-hvigdm fbhocO"
      id="categories-title"
    >
      Principais categorias
    </h2>
    <ul
      class="sc-fhHczv hNLWKe"
      role="list"
    >
      <li
        role="listitem"
      >
        <a
          aria-label="Ver categoria Eletrônicos com 5 produtos"
          class="sc-ggWZvA edjxiA"
          href="/category/1"
        >
          <article
            class="sc-dTvVRJ gybGku"
            data-testid="category-card"
          >
            <h3
              class="sc-jwTyAe kuNBmR"
            >
              Eletrônicos
            </h3>
            <p
              aria-hidden="true"
              class="sc-hjsuWn blXWuW"
            >
              5 produtos
            </p>
          </article>
        </a>
      </li>
      <li
        role="listitem"
      >
        <a
          aria-label="Ver categoria Livros com 1 produto"
          class="sc-ggWZvA edjxiA"
          href="/category/2"
        >
          <article
            class="sc-dTvVRJ gybGku"
            data-testid="category-card"
          >
            <h3
              class="sc-jwTyAe kuNBmR"
            >
              Livros
            </h3>
            <p
              aria-hidden="true"
              class="sc-hjsuWn blXWuW"
            >
              1 produto
            </p>
          </article>
        </a>
      </li>
    </ul>
  </section>
</div>
`;

exports[`<CategoriesList /> should match snapshot without categories 1`] = `
<div>
  <section
    aria-labelledby="categories-title"
    class="sc-bRKDuR GtvKZ"
    role="region"
  >
    <h2
      class="sc-hvigdm fbhocO"
      id="categories-title"
    >
      Principais categorias
    </h2>
    <p
      aria-live="polite"
      role="status"
    >
      Nenhuma categoria encontrada
    </p>
  </section>
</div>
`;
